Automattic/jetpack
# Contact Form dashboard: ensure the inbox is displayed properly for all users

## Proposed changes:

This PR introduces a few changes to the Contact Form Inbox view, available here:

<img width="1475" alt="Screenshot 2024-02-09 at 12 40 44" src="https://github.com/Automattic/jetpack/assets/426388/2bac57dd-64eb-4a49-b64f-02540b05d3eb">

1. Do not display that view and the Feedback menu to editors-
2. Do not display the Export button to non-admins.
3. Ensure the list of submissions is properly displayed in the Inbox view.

### Other information:


## Jetpack product discussion

* p1707435111778979-slack-C05PV073SG3

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

> [!WARNING]
> If you test on multisite, you will want to test with #35559, or you will not be able to connect your site to WordPress.com

* Start with a site that is connected to WordPress.com, with 3 registered users: an admin, an editor, and an author.
* Go to Jetpack > Settings and ensure that the contact form module is active.
* Go to Pages > Add Page, and publish a page with a contact form block.
* Go to the frontend, and submit a form entry.

Now, **logged in as the admin**

* You should see the Feedback menu in the admin navigation.
* In that menu, you should see the Classic <> Inbox toggle in the top right.
* In the classic view, you should see the submission you sent earlier.
* In the classic view, you should see the Export button at the top of the list table.
* In the Inbox view, you should see both the Export and the submission.

Then, **logged in as the editor**

* You should see the Feedback menu in the admin navigation.
* In that menu, you should see the Classic <> Inbox toggle in the top right.
* In the classic view, you should see the submission you sent earlier.
* In the classic view, you should **not** see the Export button at the top of the list table.
* In the Inbox view, you should see the submission.
* In the Inbox view, you should **not** see the Export button

Finally, **logged in as the author**

* You should **not** see the Feedback menu in the admin navigation.
